onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2726,14 +2760,40 @@ CUDAExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
   // These are usually shape related computation subgraphs
   // Following logic can be extended for other EPs
   auto cpu_nodes = GetCpuPreferredNodes(graph, kernel_lookup, tentative_nodes, logger);
-  std::vector<std::unique_ptr<ComputeCapability>> result;
   for (auto& node_index : candidates) {
     if (cpu_nodes.count(node_index) > 0)
       continue;
 
-    auto sub_graph = IndexedSubGraph::Create();
-    sub_graph->Nodes().push_back(node_index);
-    result.push_back(ComputeCapability::Create(std::move(sub_graph)));
+    // Previously assigned nodes have been accounted before
```

Comment:
There are some cases that an op is only implemented in CPU, or only implemented in CUDA. The capacity aware partition might need handle such case. Maybe add some TODO comments.

Reply:
The CPU preferred nodes are filtered out at this point, a few lines above, so this should not be a concern.